extensions/Corbnorb/noise.js

```diff
@@ -0,0 +1,3459 @@
+        blocks: [
+          {
+            opcode: "initNoise",
+            blockType: BlockType.COMMAND,
+            text: "create noise id:[ID] seed:[SEED] type:[TYPE] octaves:[OCTAVES] frequency:[FREQUENCY] fractal:[FRACTAL] inverted?[INVERTED] easing:[EASING]",
+            arguments: {
+              ID: {
+                type: ArgumentType.STRING,
+                defaultValue: "myNoise",
+              },
+              SEED: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+              TYPE: {
+                type: ArgumentType.STRING,
+                menu: "NOISE_TYPE",
+                defaultValue: "Perlin",
+              },
+              OCTAVES: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "1",
+              },
+              FREQUENCY: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0.01",
+              },
+              FRACTAL: {
+                type: ArgumentType.STRING,
+                menu: "FRACTAL_TYPE",
+                defaultValue: "FBm",
+              },
+              INVERTED: {
+                type: ArgumentType.STRING,
+                menu: "INVERTED_MENU",
+                defaultValue: "false",
+              },
+              EASING: {
+                type: ArgumentType.STRING,
+                menu: "EASING_TYPE",
+                defaultValue: "Linear",
+              },
+            },
+          },
+          {
+            opcode: "getNoise",
+            blockType: BlockType.REPORTER,
+            text: "get noise id:[ID] at x:[X] y:[Y] z:[Z]",
+            arguments: {
+              ID: {
+                type: ArgumentType.STRING,
+                defaultValue: "myNoise",
+              },
+              X: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+              Y: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+              Z: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "0",
+              },
+            },
+          },
+        ],
+        menus: {
+          NOISE_TYPE: {
+            acceptReporters: false,
+            items: [
+              "OpenSimpex2",
+              "OpenSimpex2S",
+              "Cellular",
+              "Perlin",
+              "Value Cubic",
+              "Value",
+            ],
+          },
+          FRACTAL_TYPE: {
+            acceptReporters: false,
+            items: ["None", "FBm", "Ridged", "Ping Pong"],
+          },
+          INVERTED_MENU: {
+            acceptReporters: true,
+            items: ["true", "false"],
+          },
+          EASING_TYPE: {
+            acceptReporters: false,
+            items: ["Linear", "Squared", "Cubed", "Root"],
+          },
+        },
+      };
+    }
+
+    initNoise(args) {
+      noises[args.ID] = [new FastNoiseLite(args.SEED), 0, 0];
```

Comment:
Please cast all your arguments, this goes for all blocks

Reply:
like set a variable for each argument?

Reply:
arguments should be casted to their respective type

Reply:
so for example if you have an argument thats a string you would do `Scratch.Cast.toString(args.MYARG)` etc

Reply:
if you reuse that argument later its easier to just do stuff like `args.MYARG = Scratch.Cast.toString(args.MYARG)`